text/0064-publishing-protobufs.md

```diff
@@ -0,0 +1,198 @@
+- Feature Name: Publishing Protobuf Definition Files
+- Start Date: 2023-03-22
+- MCIP PR: [mobilecoinfoundation/mcips#0064](https://github.com/mobilecoinfoundation/mcips/pull/0064)
+- Tracking Issue: [mobilecoinfoundation/mobilecoin/3071](https://github.com/mobilecoinfoundation/mobilecoin/issues/3071)
+
+# Summary
+[summary]: #summary
+
+A more ergonomic way for clients to use the 
+[Protobuf](https://protobuf.dev/) definition files used for communicating with
+consensus and fog services.
+
+# Motivation
+[motivation]: #motivation
+
+Protobuf definition files are often used to generate code for communicating with
+services. The Protobuf files define the binary data representation and the
+generated code provides a way to convert between a common code interface and the
+binary data representation.
+
+The current Protobuf definitions for communicating with fog and consensus are
+defined in the main 
+MobileCoin Foundation git [repo](https://github.com/mobilecoinfoundation/mobilecoin). 
+
+Most clients end up submoduling the MobileCoin Foundation repo in order to
+generate code at build time.  The MobileCoin Foundation repo is fairly
+large and submoduling it to gain access to a few files is not very ergonomic.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Protobuf definition files are made available in the following ways:
+
+1. [Buf Schema Registry](#buf-schema-registry)
+2. [Rust crates](#rust-crates)
+3. [The Protobuf git repo](#the-protobuf-git-repo)
+
+## Buf Schema Registry
+
+The [Buf Schema Registry](https://buf.build/explore/) is an online hosting
+platform of Protobuf defintions. It provides documentation hosting and a command
+line tool to detect breaking changes in Protobuf definitions.
+
+The MobileCoin Foundation Protobuf files are available at
+<https://buf.build/mobilecoinfoundation>.
+
+
+For most clients, the Buf Schema Registry is the recommended way to use the
+MobileCoin Foundation's Protobufs.
+
+## Rust Crates
+
+The MobileCoin Foundation Protobuf defintions are available via
+[Rust](https://www.rust-lang.org/) crates.  The crates use
+[prost](https://docs.rs/prost/latest/prost/) generated messages. Both
+[tonic](https://docs.rs/tonic/latest/tonic/) and
+[grpcio](https://docs.rs/grpcio/latest/grpcio/) are supported for the services.
+
+
+These crates can be found on
+<https://crates.io> by searching for one of the following:
+
+- [mc-*-grpcio](https://crates.io/search?q=mc-*-grpcio) for rust
+  [grpcio](https://docs.rs/grpcio/latest/grpcio/) services.
+- [mc-*-messages](https://crates.io/search?q=mc-*-messages) for
+  [prost](https://docs.rs/prost/latest/prost/) messages.
+- [mc-*-tonic](https://crates.io/search?q=mc-*-tonic) for
+  [tonic](https://docs.rs/tonic/latest/tonic/) services.
```

Comment:
This naming is what we currently have in https://github.com/mobilecoinfoundation/protobufs, but I'm wondering if doing a namespacing grouping might be better
```console
- mc-grpcio-*
- mc-messages-*
- mc-tonic-*
```
I also wonder about the "messages" is it too generic?